src/data_management/rasters/PaginationBar.js

```diff
@@ -2,36 +2,174 @@ import React, { Component } from "react";
+          }}
+          maxlength={(links.length + "").length}
+          size={(links.length + "").length}
+        />
+        <span> of {links.length}</span>
+      </div>
+    );
+  }
+
   render() {
     const { pages, page, loadRastersOnPage } = this.props;
 
     if (!page && !pages) {
       return null;
     }
 
+    // have not found out yet why this is wrapped in try catch block ..
     try {
-      const links = Array.from(Array(pages).keys());
-      return (
-        <div className={styles.PaginationBar}>
-          {links.map((link, i) => {
-            const linkPlusOne = link + 1;
-            if (page === linkPlusOne) {
-              return <div key={i}>{linkPlusOne}</div>;
-            }
-            return (
-              <div
-                style={{ cursor: "pointer", color: "#007bff" }}
-                key={i}
-                onClick={() => {
-                  loadRastersOnPage(linkPlusOne);
-                }}
-              >
-                <a>{linkPlusOne}</a>
-              </div>
-            );
-          })}
-        </div>
+      const showPagesAhead = 3; // amount of pages to show ahead
+      // links = array [1, ... pages]
+      const links = Array(pages)
+        .fill(0)
+        .map((e, indexx) => indexx + 1);
+      // linksStart = array [1, ... showPagesAhead]
+      const linksStart = Array(showPagesAhead)
+        .fill(0)
+        .map((e, indexx) => indexx + 1);
+      // linksCenter = [ 11,12,13,14,15,16,17] given that 14 is current page
+      const linksCenter = links.slice(
+        page - (showPagesAhead + 1),
+        page + showPagesAhead
       );
+      // linkEnd = [29,30,31,32] given that links.length =32
+      const linkEnd = links.slice(-showPagesAhead);
+      const linksFromStartToCurrentPage = links.slice(0, page + showPagesAhead);
+      const linksFromCurrentPageToEnd = links.slice(
+        page - (showPagesAhead + 1),
+        links.length
+      );
+      // holds maximum links that do not benefit from ellipsis
+      const pagesShownWithoutEllipsis = showPagesAhead * 4 + 3; // begin1+ellipsis+ before current page2+page itself+aftercurrentpage3+ellipsis+end4
+      // holds maximum distance that page is from start or end so ellipsis is not needed
+      const noEllipseBeforeOrAfterCurrentPage = showPagesAhead * 2 + 2;
+
+      // case [1,2,3,4,5,currentpage,7,8,9 .. 45,46,47]
+      // ellepsis right of current page
+      if (
+        links.length > pagesShownWithoutEllipsis &&
+        page <= noEllipseBeforeOrAfterCurrentPage
+      ) {
+        return (
+          <div>
+            <div className={styles.PaginationBar}>
+              {this.renderLinks(
+                linksFromStartToCurrentPage,
+                page,
+                loadRastersOnPage
+              )}
+              {this.renderLinks(["..."], page)}
+              {this.renderLinks(linkEnd, page, loadRastersOnPage)}
+            </div>
+            <div className={styles.NavigatorBar}>
```

Comment:
Ergens voelt het wel een beetje dubbel dat er nu 2 manieren zijn om door de pagina's heen te klikken..

Reply:
We wachten even op de rest van het product en of er nog kritiek uit de business komt